lang/java/reef-common/src/main/java/org/apache/reef/util/OSUtils.java

```diff
@@ -94,6 +95,7 @@ public static long getPID() {
             .command("powershell.exe", "-NoProfile", "-Command",
                 "wmic process where processid=$pid get parentprocessid")
             .start();
+        process.waitFor();
```

Comment:
We should probably introspect on the return value and react to cases where it isn't `0`, right?

Reply:
Fixed. Still throw -1 as previous logic because positive return code from WaitFor() may confuse with PID number.